### DISCUSSION TITLE
How librdkafka obtains consumer lag information

### DISCUSSION THREAD

**Discussion by VisonZh** [edenhill/librdkafka#3487] [Q&A] [answered]
Kafka Java Consumer API provides an interface, but the C/C++ Consumer API did not find the corresponding interface?

**Comment by VisonZh** [comment]
@edenhill

**Comment by edenhill** [comment, marked as answer]
The librdkafka consumer provides up to date consumer_lag in its metrics (https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md).

**Reply by VisonZh** [reply]
Are there other interfaces that can be called directly?

**Reply by edenhill** [reply]
Use query_watermark_offsets() (or get_watermark_offsets() for the cached version) to get the partition's high offset, then subtract the committed offset from that (committed()).

**Reply by VisonZh** [reply]
How to get the offset that has been submitted?

**Reply by edenhill** [reply]
committed()

**Reply by VisonZh** [reply]
tks！But I can't find the function in rdkafka.h

**Reply by edenhill** [reply]
https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L4073

**Reply by VisonZh** [reply]
tks very much!
